### PR TITLE
Duet3-6HC & -6XD Configs

### PR DESCRIPTION
**Pull request by ecowden**
Module: Config for Duet3-6HC and Duet3-6XD

Both the [Duet3-6HC](https://docs.duet3d.com/Duet3D_hardware/Duet_3_family/Duet_3_Mainboard_6HC_Hardware_Overview) and it's sibling, the [Duet3-6XD](https://docs.duet3d.com/Duet3D_hardware/Duet_3_family/Duet_3_Mainboard_6XD_Hardware_Overview), use the SAME70Q20B build. Both are quite similar, except that the 6XD doesn't have integrated stepper drivers. I've included both configs in one PR following the example of #5812, which adds several configs for similar boards in the BTT Manta series. I figure this makes it easier to review. Let me know if you'd prefer them separate.

I've followed formatting from the [generic-duet3-mini.cfg](https://github.com/Klipper3d/klipper/blob/master/config/generic-duet3-mini.cfg), straying only when I found minor inconsistencies and felt like it was appropriate to make a call. Feedback welcome.

I've added both to `printers.test` and confirmed that the tests pass on a Raspberry Pi 4.

For pin references, see the open source schematic PDFs for the [6HC v1.02](https://github.com/Duet3D/Duet3-Mainboard-6HC/tree/master/Duet3_Mainboard_6HC_v1.02) and [6XD v1.0](https://github.com/Duet3D/Duet3-Mainboard-6XD/tree/master/v1.0), respectively.

Hat tip to @lukeslaboratory, who's PR #5818 got me started. 🎩 

Signed-off-by: Evan Cowden [evan38109@gmail.com](mailto:evan38109@gmail.com)

### CODE REVIEW

File: config/generic-duet3-6xd.cfg [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# This file contains common pin mappings for the Duet3 6HC. To use
```
Comment (Wulfsta) [inline config/generic-duet3-6xd.cfg]: Typo in the comment, this should be 6XD.

**Review by JamesH1978 — APPROVED** [review]
Looks fine to me

Thanks
James

**Comment by KevinOConnor** [comment]
Thanks.

-Kevin

**Comment by HereticDezign** [comment]
@ecowden 

Thank you for this, but... im sorry there appears to be a "typo" in the information section of the 6HC example config.
GPIO-out I07 is listed as using pin PD23 
This is INCORRECT, as the correct pin is actually PC23.

Out5. Tach Uses PD23.

I have checked and cross referenced this using the Duet PDF file from the following link:

https://github.com/Duet3D/Duet3-Mainboard-6HC/blob/master/Duet3_Mainboard_6HC_v1.02/Duet3_MB_6HC_Schematic_v1.02.pdf

**Pull request merged**